django/middleware/csrf.py

```diff
@@ -14,7 +14,8 @@
 from django.utils.cache import patch_vary_headers
 from django.utils.crypto import constant_time_compare, get_random_string
 from django.utils.encoding import force_text
-from django.utils.http import same_origin
```

Comment:
Worth noting: `same_origin` was literally only used inside here, for all of the codebase.

But since it's technically sorta part of the public API of the `django.utils.http` package, I left it. Not sure if it's just safe to remove this function or not.